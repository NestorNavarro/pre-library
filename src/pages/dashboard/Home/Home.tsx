import { useForm, useWatch }                                  from "react-hook-form";
import { Stack }                                              from "@mantine/core";

import FormProvider from "../../../components/global/hook-form/FormProvider";
// import RHFPasswordInput from "../../../components/global/hook-form/RHFPasswordInput";
import RHFSwitch from "../../../components/global/hook-form/RHFSwitch";
import RHFTextArea from "../../../components/global/hook-form/RHFTextArea";
import RHFTextInput from "../../../components/global/hook-form/RHFTextInput";

const Home = () => {
	const methods = useForm({
		defaultValues : {
			inpt : "",
		},
	});

	const { handleSubmit, control } = methods;

	const watch = useWatch({ name : "inpt", control });
	console.log(watch);

	const onSubmit = (data : {}) => console.log(data);

	const onChangeText = (value : any) => {
		console.log(value);
	};

	return (
		<FormProvider methods={methods} onSubmit={handleSubmit(onSubmit)}>
			<Stack>
				{/* We need to define the inpt in default value to avoid the uncorntoler warning */}
				{/*TODO: Testing onChange */}
				<RHFTextInput name="inpt" label="InptText" onChange={onChangeText}/>
				<RHFTextArea name="textArea" label="TextArea" onChange={onChangeText} />
				<RHFSwitch name="switch" label="Switch" />
				{/* <RHFPasswordInput name="password" label="password "/> */}
			</Stack>
		</FormProvider>
	);
};

export default Home;
