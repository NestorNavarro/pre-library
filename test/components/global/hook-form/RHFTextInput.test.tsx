/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from '@testing-library/react';
import Home from "../../../../src/pages/dashboard/Home"

describe("RHFTextInput", () => { 
    test("should pass", () => { 
        const { container } = render(<Home />);
        expect( container ).toMatchSnapshot();
    });
});