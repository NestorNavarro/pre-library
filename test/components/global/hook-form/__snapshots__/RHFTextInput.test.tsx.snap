// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RHFTextInput should pass 1`] = `
<div>
  <form>
    <div
      class="mantine-Stack-root mantine-lfk3cq"
    >
      <div
        class="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
      >
        <label
          class="undefined mantine-InputWrapper-label mantine-TextInput-label mantine-ul5p4k"
          for="mantine-r0"
          id="mantine-r0-label"
        >
          InptText
        </label>
        <div
          class="undefined mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            class="mantine-Input-input mantine-TextInput-input mantine-yko1sf"
            id="mantine-r0"
            name="inpt"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="mantine-InputWrapper-root mantine-Textarea-root mantine-1m3pqry"
      >
        <label
          class="undefined mantine-InputWrapper-label mantine-Textarea-label mantine-ul5p4k"
          for="mantine-r1"
          id="mantine-r1-label"
        >
          TextArea
        </label>
        <div
          class="mantine-Input-wrapper mantine-Textarea-wrapper mantine-12sbrde"
        >
          <textarea
            aria-invalid="false"
            class="mantine-Input-input mantine-Textarea-input mantine-12rr3iy"
            id="mantine-r1"
            name="textArea"
          />
        </div>
      </div>
      <div
        class="mantine-Switch-root mantine-15po0m8"
      >
        <div
          class="mantine-15e18iv mantine-Switch-body"
        >
          <input
            class="mantine-1buhfoz mantine-Switch-input"
            id="switch"
            name="switch"
            type="checkbox"
          />
          <label
            class="mantine-1pauhrz mantine-Switch-track"
            for="switch"
          >
            <div
              class="mantine-11dx59s mantine-Switch-thumb"
            />
            <div
              class="mantine-69c9zd mantine-Switch-trackLabel"
            />
          </label>
          <div
            class="mantine-1xy33qi mantine-Switch-labelWrapper"
          >
            <label
              class="mantine-1jc5t4g mantine-Switch-label"
              for="switch"
            >
              Switch
            </label>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
